src/main/pages/developer-guide/databases.adoc

```diff
@@ -0,0 +1,177 @@
+
+Exception handler has to be defined in persistence XML.
+
+[id="test-compatibility-kit"]
+== Test Compatibility Kit (TCK)
+Test Compatibility kit is a set of integration tests that covers uses of essential DAO components, that are being run against real database.
+
+List of Che modules that have TCK tests:
+
+* che-core-api-user
+* che-core-api-account
+* che-core-api-installer
+* che-core-api-ssh
+* che-core-api-workspace
+* che-core-api-workspace-activity
+* infrastructure-kubernetes
+* che-multiuser-machine-authentication (Multi User Che)
+* che-multiuser-api-organization (Multi User Che)
+* che-multiuser-api-permission (Multi User Che)
+* che-multiuser-api-resource (Multi User Che)
+* che-multiuser-permission-workspace (Multi User Che)
+
+Tests in these modules are executed on H2 Database during the build.
+Also, tests themselves are packaged into separate artifact with `tests` classifier, which are meant to be used for their re-usage against other DBs.
+
+As an example of that, there are `postgresql-tck` and `che-multiuser-postgresql-tck` modules for running TCK tests on Postgres DB.
+
+[id="cascade-removal-tests"]
+== JPA Cascade Removal Tests
+
+In Eclipse Che database schema, some objects have foreign keys reference on other objects.
+Example: Account and workspaces, Permissions and Workspaces. To be able to correctly
+remove such objects and avoid referential integrity problems Eclipse Che are using events mechanism.
+For example, when Account DAO wants to remove account it sends CascadeEvent, and Workspace DAO has a CascadeEventSubscriber that would listen to that event
+and remove workspace at that time. In case if removal of workspace failed the whole transaction would be rolled back.
+
+All that functionality is covered by "Cascade Removal Tests", that are located in `cascade-removal` and `che-multiuser-cascade-removal` modules.
+
+[id="flyway-migration"]
+== Flyway migration
+
+Flyway framework is used to facilitate database schema initialization, as well as perform migrations.
+It is configured to pick up scripts and apply them in order, according to versioning.
+Some key configuration values are defined in che.properties:
+
+
+----
+db.schema.flyway.baseline.enabled=true                      #
+db.schema.flyway.baseline.version=5.0.0.8.1                 #
+db.schema.flyway.scripts.prefix=                            #
+db.schema.flyway.scripts.suffix=.sql                        #
+db.schema.flyway.scripts.version_separator=__               #
+db.schema.flyway.scripts.locations=classpath:che-schema     #
+----
+
+Che DB Schema artifacts have
+
+Taking a look at the sample structure:
+
+----
+che-schema/
+    5.0.0-M1/
+        1__init.sql
+    5.0.0-M2
+        1__rename_tables.sql
+        2__insert_predefined_data.sql
+        postgresql/
+            2__insert_predefined_data.sql
+subproject-schema/
+    5.0.0-M1/
+        1.1__init_subproject.sql
```

Comment:
It makes sense to give here an example with `0.1_before_main_init.sql`

Reply:
I don't think the content of the script matters

Reply:
It would show that `0` can be used if there is a need to perform script before first in super schema.

Reply:
oh, ok